arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ResourceTest.kt

```diff
@@ -443,6 +453,44 @@ class ResourceTest : ArrowFxSpec(
         r.asFlow().map { it + 1 }.toList() shouldBe listOf(n + 1)
       }
     }
+
+    suspend fun checkAllocated(mkResource: (() -> Int, (Int, ExitCase) -> Unit) -> Resource<Int>) {
+      val released = CompletableDeferred<Int>()
+      val releaseValue = Random.nextInt()
+      val seed = Random.nextInt()
+
+      val (allocate, release) = mkResource({ seed }) { _, _ -> released.complete(releaseValue) }.allocated()
+
+      allocate() shouldBe seed
+
+      release(1, ExitCase.Completed)
```

Comment:
We're only testing the happy path with `ExitCase.Completed`, we should also test `ExitCase.Failed` and `ExitCase.Cancelled` code branches.

Reply:
I've updated the tests to call release with each of the three exist cases,  although I'm not sure that I'm really testing the error path effectively.